### DISCUSSION TITLE
How to set custom headers for websocket?

### DISCUSSION THREAD

**Discussion by mokhtarabadi** [eycorsican/leaf#42] [Q&A] [answered]
because sometimes I need to give the leaf IP instead of the hostname, I need to set the Host header in WebSocket (because I used CDN)

**Comment by eycorsican** [comment, marked as answer]
The feature is supported on master now. https://github.com/eycorsican/leaf/blob/master/README.zh.md#ws-1

**Reply by mokhtarabadi** [reply]
Thanks so much

**Comment by qiuzi** [comment]
提示udp downlink error: ws error: UTF-8 encoding error

**Reply by eycorsican** [reply]
服务端实现有问题，或者正确点来说，跟这个项目不兼容。服务端实现不应该把 binary 数据放进 ws 的 text frame 中发过来。
